### PR TITLE
Refactored `BaseConsole::updateProgress()`

### PR DESCRIPTION
**Pull request by developeruz**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | 

Because `BaseConsole` has `F` rating in codeclimate

### CODE REVIEW

File: framework/helpers/BaseConsole.php [hunk 1]
```diff
@@ -957,46 +957,12 @@ public static function startProgress($done, $total, $prefix = '', $width = null)
...
-            }
-        }
-        if ($prefix === null) {
-            $prefix = self::$_progressPrefix;
-        } else {
-            self::$_progressPrefix = $prefix;
-        }
-        $width -= static::ansiStrlen($prefix);
-
+        $prefix = $prefix === null ? self::$_progressPrefix : self::$_progressPrefix = $prefix;
```
Comment (samdark) [inline framework/helpers/BaseConsole.php]: This one-liner is quite complicated with both comparisons and assignments. I'd prefer if-structure instead exactly as it was before.

Additionally, new code does one extra weird assignment: `$prefix = self::$_progressPrefix = $prefix;`.

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/helpers/BaseConsole.php [hunk 2]
```diff
@@ -1022,6 +988,52 @@ public static function updateProgress($done, $total, $prefix = null)
         flush();
     }
 
+    /**
+     * @param $prefix
```
Comment (samdark) [inline framework/helpers/BaseConsole.php]: phpdoc missing.

File: framework/helpers/BaseConsole.php [hunk 3]
```diff
@@ -1022,6 +988,52 @@ public static function updateProgress($done, $total, $prefix = null)
...
+            $width = floor($screenSize[0] * $width);
+        }
+
+        $width -= static::ansiStrlen($prefix);
+
+        return $width;
+    }
+
+    /**
+     * @param $done
```
Comment (samdark) [inline framework/helpers/BaseConsole.php]: phpdoc missing

File: framework/helpers/BaseConsole.php (hunk 1 above)
Comment (developeruz) [reply framework/helpers/BaseConsole.php]: Fixed

File: framework/helpers/BaseConsole.php [hunk 4]
```diff
@@ -1022,6 +992,60 @@ public static function updateProgress($done, $total, $prefix = null)
         flush();
     }
 
+    /**
+     * Return screen width
```
Comment (samdark) [inline framework/helpers/BaseConsole.php]: It is not screen width. Screen width was returned by `static::getScreenSize()` inside.

**Comment by samdark** [comment]
Other than method naming it looks good to me. @developeruz do you have an idea of how to name this method better?

**Comment by developeruz** [comment]
Renamed to `getProgressbarWidth`, based on [`self::$_progressWidth`](https://github.com/developeruz/yii2/blob/0eaa6a3599a7e31ebb26486f3d08769131bb5f73/framework/helpers/BaseConsole.php#L926) description

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**